Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java

```diff
@@ -2021,6 +2048,178 @@ public ImageView.ScaleType getScaleTypes(String scale){
+             layout.removeAllViews();
+             layout.addView(invoiceLayout);
+        } else {
+            
+            requestCameraPermission(() -> renderCameraProfilePicture(id));
+       
+            
+        }
+
+             
+        });
+    }
+
+private boolean isCameraPermissionGranted() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        int cameraPermission = ContextCompat.checkSelfPermission(context, Manifest.permission.CAMERA);
+        return cameraPermission == PackageManager.PERMISSION_GRANTED;
+    }
+    // For devices below Android Marshmallow, the permission is granted by default at installation time
+    return true;
+}
+
+private void requestCameraPermission(Runnable callback) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        //Toast.makeText(activity, "Need Permission To Access The Camere", Toast.LENGTH_SHORT).show();
+        ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.CAMERA}, IMAGE_PERMISSION_REQ_CODE_PROFILE);
+        // Save the callback to execute after permission is granted
+        cameraPermissionCallback = callback;
+    }
+}
+
+
+
+
+    Executor getExecutor() {
+        return ContextCompat.getMainExecutor(activity);
+    }
+
+@SuppressLint("RestrictedApi")
+    private void startCameraX(ProcessCameraProvider cameraProvider) {
+        cameraProvider.unbindAll();
+        CameraSelector cameraSelector = new CameraSelector.Builder()
+                .requireLensFacing(CameraSelector.LENS_FACING_FRONT)
+                .build();
+        Preview preview = new Preview.Builder()
+                .build();
+        preview.setSurfaceProvider(previewView.getSurfaceProvider());
+
+        imageCapture = new ImageCapture.Builder()
+                .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+                .build();
+        ImageAnalysis imageAnalysis = new ImageAnalysis.Builder()
+                .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                .build();
+
+        imageAnalysis.setAnalyzer(getExecutor(), this::analyze);
+
+        cameraProvider.bindToLifecycle((LifecycleOwner) activity, cameraSelector, preview, imageCapture);
+    }
+
+
+    public void analyze(@NonNull ImageProxy image) {
+        Log.d("TAG", "analyze: got the frame at: " + image.getImageInfo().getTimestamp());
+        image.close();
+    }
+
+private void capturePhoto() {
+        long timestamp = System.currentTimeMillis();
+
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(MediaStore.MediaColumns.DISPLAY_NAME, timestamp);
+        contentValues.put(MediaStore.MediaColumns.MIME_TYPE, "image/jpeg");
+        imageCapture.takePicture(
+                new ImageCapture.OutputFileOptions.Builder(context.getContentResolver(), MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentValues).build(),
+                getExecutor(),
```

Comment:
No need to call this function, directly do `ContextCompat.getMainExecutor(activity)`

Reply:
didn't get it

Reply:
simple replace this line with `ContextCompat.getMainExecutor(activity)`